#! /bin/sh
patch -p1 -l -f $* < $0
exit $?


From: Albert Astals Cid <aacid@kde.org>
Date: Mon, 31 Mar 2025 14:35:49 +0200
Subject: [PATCH 4/7] Move isOk check to inside JBIG2Bitmap::combine

(cherry picked from commit 1f151565bbca5be7449ba8eea6833051cc1baa41)

CVE-2025-32365
---
 poppler/JBIG2Stream.cc | 9 +++++----
 1 file changed, 5 insertions(+), 4 deletions(-)

diff --git a/poppler/JBIG2Stream.cc b/poppler/JBIG2Stream.cc
index a6dba047..9e8d0cc9 100644
--- a/poppler/JBIG2Stream.cc
+++ b/poppler/JBIG2Stream.cc
@@ -771,6 +771,9 @@ void JBIG2Bitmap::combine(JBIG2Bitmap *bitmap, int x, int y, unsigned int combOp
     unsigned int src0, src1, src, dest, s1, s2, m1, m2, m3;
     bool oneByte;

+    if (unlikely(!isOk())) {
+        return;
+    }
     // check for the pathological case where y = -2^31
     if (y < -0x7fffffff) {
         return;
@@ -2202,9 +2205,7 @@ void JBIG2Stream::readTextRegionSeg(unsigned int segNum, bool imm, bool lossless
             if (pageH == 0xffffffff && y + h > curPageH) {
                 pageBitmap->expand(y + h, pageDefPixel);
             }
-            if (pageBitmap->isOk()) {
-                pageBitmap->combine(bitmap.get(), x, y, extCombOp);
-            }
+            pageBitmap->combine(bitmap.get(), x, y, extCombOp);

             // store the region bitmap
         } else {
--
2.47.2
